app/jobs/drive_application_job.rb

```diff
@@ -0,0 +1,7 @@
+class DriveApplicationJob < ApplicationJob
+  def perform(snap_application:)
+    if snap_application.driver_applications.empty?
+      MiBridges::Driver.new(snap_application: snap_application).run
+    end
```

Comment:
I was just thinking about this.  Instead of us having two approaches to driving any application, those being:

1. Drive for the first time. IE: `.run`
2. Re-drive / re-run a drive. IE: `.re_run`

What do you think about changing it to push the logic down into the driver where:

1. If a driver_application exists - re-run using the log in flow
2. If no driver_application exists - run the sign-up flow

Reply:
That sounds right -- I originally wanted to have them be entirely different because it seemed simpler to ensure that `run` can only happen once, but that isn't really happening so we may as well add some smart logic. Should I do that via a separate PR?

Reply:
Should also add that this approach would not change the fact that we'd need logic in here to not run or re run, since we don't want to re-attempt a drive or re-drive just because someone reloads the page

Reply:
> Should also add that this approach would not change the fact that we'd need logic in here to not run or re run, since we don't want to re-attempt a drive or re-drive just because someone reloads the page

I think we might still be able to make it idempotent. Like say we did it like:

1. find or create the first driver application for a given snap application.
2. enqueue the job with that driver application
3. add a "driven_at" timestamp column that's nil by default
4. When the job is run, 
   * if `driven_at` is nil, set the timestamp, and run it.
   * if `driven_at` is some time in the last 24 hours, don't run it.

Reply:
ok yeah, or really if there is *any* timestamp we wouldnt want to re-run, since re-running will be an admin only thing.

is this a change you want to see in this PR? I am looking for which feedback you want for this to get merged :)

Reply:
All this can be totally in for further PR's